alf/algorithms/entropy_target_algorithm.py

```diff
@@ -176,99 +163,105 @@ def train_step(self, distribution, step_type):
                     extra=EntropyTargetLossInfo(neg_entropy=-entropy))))
 
     def calc_loss(self, training_info: EntropyTargetInfo, valid_mask=None):
+        """Calculate loss.
+
+        Args:
+            training_info (EntropyTargetInfo): for computing loss.
+            valid_mask (tensor): valid mask to be applied on time steps.
+        
+        Returns:
+            LossInfo.
+        """
         loss_info = training_info.loss
-        mask = tf.cast(training_info.step_type != StepType.LAST, tf.float32)
+        mask = (training_info.step_type != StepType.LAST).type(torch.float32)
         if valid_mask:
-            mask = mask * tf.cast(valid_mask, tf.float32)
+            mask = mask * (valid_mask).type(torch.float32)
         entropy = -loss_info.extra.neg_entropy * mask
-        num = tf.reduce_sum(mask)
+        num = torch.sum(mask)
         not_empty = num > 0
-        num = tf.maximum(num, 1)
-        entropy2 = tf.reduce_sum(tf.square(entropy)) / num
-        entropy = tf.reduce_sum(entropy) / num
-        entropy_std = tf.sqrt(tf.maximum(0.0, entropy2 - entropy * entropy))
-
-        run_if(not_empty, lambda: self.adjust_alpha(entropy))
+        num = max(num, 1)
+        entropy2 = torch.sum(entropy**2) / num
```

Comment:
suggest using math_ops.square because **2 is much slower than pow():https://github.com/pytorch/pytorch/issues/2618